File: Day6Part1.java
import java.util.List;

public class Day6Part1 {
    public static void main(String[] args) {

        long count = new GroupingParser()
            .stream("input-day6.txt")
            .mapToLong(Day6Part1::answers)
            .reduce(0, (a,b) -> a + b);

        System.out.println(count);
    }

    static long answers(List<String> group) {
        return String.join("", group).chars().distinct().count();
    }
}

File: day6part1.clj
(defn group [strs]
  (loop [ss strs current [] groups []]
    (let [s (first ss)]
      (if (empty? ss)
        (conj groups current)
        (if (empty? s)
          (recur (rest ss) [] (conj groups current))
          (recur (rest ss) (conj current s) groups))))))

(println
  (reduce +
    (map 
      (fn [group] (count (distinct (reduce concat group))))
      (group (.split (slurp "input-day6.txt") "\n")))))
